src/classes/EscalationTeamTrigger.cls

```diff
@@ -0,0 +1,140 @@
+/**
+* Trigger work for escalation teams
+*
+* @author Preeti (ppreeti@redhat.com)
+* @testClasses EscalationTeamTrigger_Test
+*/
+public with sharing class EscalationTeamTrigger extends BaseTrigger {
+	/** The old map of objects */
+	@TestVisible Map<Id, rh_escal__EscalationTeam__c> oldMap {
+		get {
+			return (Map<Id, rh_escal__EscalationTeam__c>) ((BaseTrigger) this).baseOldMap;
+		}
+		set;
+	}
+
+	/** The new map of objects */
+	@TestVisible Map<Id, rh_escal__EscalationTeam__c> newMap {
+		get {
+			if (this.newMap == null) {
+				this.newMap = new Map<Id, rh_escal__EscalationTeam__c>();
+
+				if (!this.isBefore && !this.newObjs.isEmpty()) {
+					this.newMap = new Map<Id, rh_escal__EscalationTeam__c>(this.newObjs);
+				}
+			}
+
+			return this.newMap;
+		}
+		set;
+	}
+
+	/** The new list of objects */
+	@TestVisible List<rh_escal__EscalationTeam__c> newObjs {
+		get {
+			return (List<rh_escal__EscalationTeam__c>) ((BaseTrigger) this).baseNewObjs;
+		}
+		set;
+	}
+
+	/**
+	* The constructor
+	*
+	* @param xoldMap The old map from the trigger
+	* @param xnewObj The list of new objects from the trigger
+	* @param isBefore If the trigger is in before or after
+	*/
+	public EscalationTeamTrigger(Map<Id, rh_escal__EscalationTeam__c> xoldMap, List<rh_escal__EscalationTeam__c> xnewObjs, Boolean isBefore) {
+		super(xoldMap, xnewObjs, isBefore);
+	}
+
+	/**
+	* Sets the name of the trigger
+	*
+	* @return The name of the trigger
+	*/
+	public override String getTriggerName() {
+		return 'EscalationTeamTrigger';
+	}
+
+	/**
+	* Do not allow addition of duplicate team members on an escalation
+	*/
+	@TestVisible
+	private void teamDupCheck() {
+		if (this.isDelete) {
+			return;
+		}
+
+		Set<Id> escalationIds = new Set<Id>();
+
+		for (rh_escal__EscalationTeam__c team: this.newObjs) {
+			escalationIds.add(team.rh_escal__Escalation__c);
+		}
+
+		escalationIds.remove(null);
+
+		if (escalationIds.isEmpty()) {
+			return;
+		}
+
+		Map<Id, Set<Id>> escalToMembers = new Map<Id, Set<Id>>();
+		Map<Id, Set<String>> escalToEmails = new Map<Id, Set<String>>();
+
+		for (rh_escal__EscalationTeam__c team: EscalationTeamUtils.getEscalationTeamMembers(escalationIds)) {
+			if (!escalToMembers.containsKey(team.rh_escal__Escalation__c)) {
```

Comment:
Remove this and build up the map outside of the loop.  This will mean we don't have to do a null check below

```java
for (Id escalationId : escalationIds) {
    escalToMembers.put(escalationId, new Set<Id>());
    escalToEmail.put(escalationId, new Set<Id>());
}
```